src/main/java/org/springframework/hateoas/mvc/ControllerLinkBuilder.java

```diff
@@ -53,17 +50,21 @@
  * @author Kevin Conaway
  * @author Andrew Naydyonock
  * @author Oliver Trosien
+ * @author Josh Ghiloni
+ * @author Greg Turnquist
  */
 public class ControllerLinkBuilder extends LinkBuilderSupport<ControllerLinkBuilder> {
 
 	private static final String REQUEST_ATTRIBUTES_MISSING = "Could not find current request via RequestContextHolder. Is this being called from a Spring MVC handler?";
-	private static final CachingAnnotationMappingDiscoverer DISCOVERER = new CachingAnnotationMappingDiscoverer(

```

Comment:
The issue is all the static fields. The solution was to have a setter method such that it would fetch a default static, unless the setter had overridden a field.

Moving things around, I was able to put the caching back inside CachingANnotationMappingDiscoverer, but at that point, I ended up STILL fetching the final route after property resolution before caching. So I don't see how to do this.